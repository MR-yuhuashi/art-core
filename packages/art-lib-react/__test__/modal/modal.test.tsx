/**
 * 插入位置（传不传parentSelector）
 * 绑定正确的class name
 */
import React from 'react';
import Modal from '../../src/components/modal/modal';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

const wrapper = mount(
  <Modal
    isOpen={true}
    className={'test-modal'}>
    <p>123</p>
    <span>dddd</span>
  </Modal>
);

const hideWrapper = mount(
  <Modal
    isOpen={false}
    className={'test-modal'}>
    <p>123</p>
    <span>dddd</span>
  </Modal>
);

test('open and close', (() => {
  expect(wrapper.hasClass('test-modal')).toBeTruthy();
  expect(hideWrapper.hasClass('test-modal')).toBeTruthy();
  expect(hideWrapper.exists('UnTouchable')).toBeFalsy();
  expect(wrapper.exists('UnTouchable')).toBeTruthy();
  expect(toJson(wrapper)).toMatchSnapshot();
  expect(toJson(hideWrapper)).toMatchSnapshot();
}));