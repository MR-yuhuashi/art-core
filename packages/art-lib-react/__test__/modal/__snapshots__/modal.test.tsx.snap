// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open and close 1`] = `
<Modal
  bodyOpenClassName="modal-portal-body-open"
  className="test-modal"
  closeTimeoutMS={400}
  isOpen={true}
  mask={true}
  parentSelector={[Function]}
  portalClassName="modal-portal"
  portalType="normal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {},
      "overlay": Object {},
    }
  }
>
  <Portal
    containerInfo={
      <div
        class="modal-portal test-modal active"
        style="z-index: 100; content: [object Object];"
        v-modal-portal="normal"
      >
        <div
          v-un-touchable=""
        >
          <div
            class="modal-portal-overlay modal-portal-overlay-after-open"
          />
          <div
            class="modal-portal-wrap"
          >
            <div
              class="modal-portal-content modal-portal-content-after-open"
            >
              <p>
                123
              </p>
              <span>
                dddd
              </span>
            </div>
          </div>
        </div>
      </div>
    }
  >
    <ModalPortal
      bodyOpenClassName="modal-portal-body-open"
      className="test-modal"
      closeTimeoutMS={400}
      defaultStyles={
        Object {
          "content": Object {
            "WebkitOverflowScrolling": "touch",
          },
          "overlay": Object {},
        }
      }
      isOpen={true}
      mask={true}
      onAfterClosed={[Function]}
      onBeforeOpen={[Function]}
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {},
          "overlay": Object {},
        }
      }
    >
      <UnTouchable>
        <div
          v-un-touchable=""
        >
          <FastClick
            className="modal-portal-overlay modal-portal-overlay-after-open"
            onClick={[Function]}
            style={Object {}}
            threshold={15}
            timeThreshold={200}
          >
            <div
              className="modal-portal-overlay modal-portal-overlay-after-open"
              onClick={[Function]}
              onTouchCancel={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
            />
          </FastClick>
          <div
            className="modal-portal-wrap"
            style={Object {}}
          >
            <div
              className="modal-portal-content modal-portal-content-after-open"
              onKeyDown={[Function]}
            >
              <p>
                123
              </p>
              <span>
                dddd
              </span>
            </div>
          </div>
        </div>
      </UnTouchable>
    </ModalPortal>
  </Portal>
</Modal>
`;

exports[`open and close 2`] = `
<Modal
  bodyOpenClassName="modal-portal-body-open"
  className="test-modal"
  closeTimeoutMS={400}
  isOpen={false}
  mask={true}
  parentSelector={[Function]}
  portalClassName="modal-portal"
  portalType="normal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {},
      "overlay": Object {},
    }
  }
>
  <Portal
    containerInfo={
      <div
        class="modal-portal test-modal"
        style="z-index: 101; content: [object Object];"
        v-modal-portal="normal"
      />
    }
  >
    <ModalPortal
      bodyOpenClassName="modal-portal-body-open"
      className="test-modal"
      closeTimeoutMS={400}
      defaultStyles={
        Object {
          "content": Object {
            "WebkitOverflowScrolling": "touch",
          },
          "overlay": Object {},
        }
      }
      isOpen={false}
      mask={true}
      onAfterClosed={[Function]}
      onBeforeOpen={[Function]}
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {},
          "overlay": Object {},
        }
      }
    />
  </Portal>
</Modal>
`;
